python/paddle/fluid/dygraph/dygraph_to_static/loop_transformer.py

```diff
@@ -111,15 +125,29 @@ def visit_Name(self, node):
             self.in_loop_vars[loop_node].add(node)
         self.generic_visit(node)
 
+    def visit_Attribute(self, node):
+        attr_full_name = get_attribute_full_name(node)
+        if callable(attr_full_name):
```

Comment:
Type of `attr_full_name ` is `str` and `callable(str)` will always return `False`. Is this as expected？

Reply:
My fault, should be eval(attr_full_name)

Reply:
Found that eval(attr_full_name) may also have problem such as attr_full_name = "self.XXX", please check my new solution, which checks whether the gast.Name/gast.Attribute node is at `func` field of gast.Call